packages/webviz-core/src/panels/Teleop/index.js

```diff
@@ -0,0 +1,291 @@
+  };
+
+  static getDerivedStateFromProps(props: Props, state: PanelState) {
+    const newState: $Shape<PanelState> = parseInput(props.config.value);
+    let changed = false;
+
+    if (props !== state.cachedProps) {
+      newState.cachedProps = props;
+      changed = true;
+    }
+
+    if (props.datatypes !== state.cachedProps.datatypes) {
+      newState.datatypeNames = Object.keys(props.datatypes).sort();
+      changed = true;
+    }
+
+    return changed ? newState : null;
+  }
+
+  composeTwist = (key) => {
+    const COMMANDS = { // Map x, y, z, th here
+      ArrowUp: [1, 0, 0, 0], 
+      ArrowDown: [-1, 0, 0, 0], 
+      ArrowLeft: [0, 0, 0, 1], 
+      ArrowRight: [0, 0, 0, -1],
+      " ": [0, 0, 0, 0], // Stop
+    };
+    const [x, y, z, th] = COMMANDS[key];
+    const speed = 0.5, turn = 1.0;
+    return {
+      linear: {
+        x: x*speed,
+        y: y*speed,
+        z: z*speed
+      },
+      angular: {
+        x: 0,
+        y: 0,
+        z: th * turn
+      }
+    }
+  };
+
+  _onChange = (event: SyntheticInputEvent<HTMLTextAreaElement>) => {
+    this.props.saveConfig({ value: event.target.value });
+  };
+
+  _onCommandButtonClick = (command) => {
+    // Simulate key press and release
+    this._handleKey({key: command, type: "keydown"});
+    this._handleKey({key: command, type: "keyup"});
+  };
+
+  _handleKey = (event) => {
+    this.setState(function(state, props) {
+      let newPressing = state.pressing;
+      newPressing[event.key] = event.type == "keydown";
+      if(this._publisher.current){
+        if(event.type=="keydown" ){
+          this._publisher.current.publish(this.composeTwist(event.key));
+        }
+      } else {
+        console.error("Publisher not set!");
+      }
+      return {
+        pressing: newPressing
+      };
+    });
+
+    this.forceUpdate(); // HACK Appears to be required to update the colors quickly???
+  };
+
+  // Add handler here for all desired keybindings (they are all the same!)
+  _keyHandlers = {
+    ArrowLeft: this._handleKey,
+    ArrowRight: this._handleKey,
+    ArrowUp: this._handleKey,
+    ArrowDown: this._handleKey,
+    " ": this._handleKey, // Space
+  };
+
+  _renderMenuContent() {
+    const { config, saveConfig } = this.props;
+
+    return (
+      <>
+        <Item>
+          <PanelToolbarLabel>Button color (rgba or hex)</PanelToolbarLabel>
+          <PanelToolbarInput
+            type="text"
+            value={config.buttonColor}
+            onChange={(event) => {
+              saveConfig({ buttonColor: event.target.value });
+            }}
+            placeholder="rgba(1,1,1,1) or #FFFFFF"
+          />
+        </Item>
+        <Item>
+          <PanelToolbarLabel>Topic name</PanelToolbarLabel>
+          <PanelToolbarInput
+            type="text"
+            value={config.topicName}
+            onChange={(event) => {
+              saveConfig({ topicName: event.target.value });
```

Comment:
Would need some data validation to ensure the saved topic name is valid

Reply:
Not sure about what would be required to validate the topic. Would it require checking that the topic name exists and it's of the correct datatype? Doing that would limit use a lot, I think, since it may very well happen that the topic doesn't "exist" since no other nodes have ever published to it, and the Teleop panel will be the only one publishing. Any pointers? 
Perhaps a) if it doesn't exist, just accept it, and b) if it already exists, ensure that the datatype is actually "geometry_msgs/Twist"?

Reply:
Sorry about the late reply. A simple validation like this should be ok:
- name is not empty 
- name should start with `/`
- name should be at least 2 characters 

You could refer to some of the validators here: https://github.com/cruise-automation/webviz/blob/master/packages/webviz-core/shared/validators.js#L102-L107

Reply:
I've added some validation on 671f133. It shows an error message on the panel if those conditions are not met. It also disables all the buttons and keypresses do nothing.